iaso/utils/models/encrypted_text_field.py

```
@@ -0,0 +1,32 @@
+import typing
+
+from cryptography.fernet import Fernet
+from django.conf import settings
+from django.db import models
+
+
+class EncryptedTextField(models.TextField):
+    description = "A Fernet-encrypted text field"
+
+    def from_db_value(
+        self,
+        value: typing.Optional[str],
+        expression: typing.Optional[str],
+        connection: typing.Any,
+    ) -> typing.Optional[str]:
+        if value is None:
+            return value
+
+        return Fernet(settings.ENCRYPTED_TEXT_FIELD_KEY).decrypt(value.encode("utf-8")).decode("utf-8")
```

I would maybe raise an exception if `settings.ENCRYPTED_TEXT_FIELD_KEY` is not defined or not in the right format, but this is probably not the right place to do that

---

FYI: I'm reusing a similar setup in Pegasus, I added this check in the Django app config, inside the `ready()` method

---

Hm I see why you'd want this, but not sure where this should be added in IASO :thinking:

---

I'll ping you once the Pegasus PR is ready

---

You can check https://github.com/BLSQ/console/pull/1/ - `apps/utils/apps.py`